types/jss/css.d.ts

```diff
@@ -0,0 +1,130 @@
+// These CSS typings adapted from TypeStyle: https://github.com/typestyle/typestyle
+
+import { Observable } from './observable'
+import * as csstype from 'csstype'
+
+export type ObservableProperties<P> = {
+  [K in keyof P]: P[K] | Observable<P[K]>
+}
+
+export type CSSProperties =
+	& ObservableProperties<csstype.Properties<string | number>>
+	& ObservableProperties<csstype.PropertiesHyphen<string | number>>;
+
+export interface JssProps {
+	'@global'?: CSSProperties;
+	extend?: string;
+	composes?: string | string[];
+}
+
+export interface JssExpand {
+	animation:
+		| {
+				delay: CSSProperties['animationDelay'];
+				direction: CSSProperties['animationDirection'];
+				duration: CSSProperties['animationDuration'];
+				iterationCount: CSSProperties['animationIterationCount'];
+				name: CSSProperties['animationName'];
+				playState: CSSProperties['animationPlayState'];
+				timingFunction: any;
+			}
+		| CSSProperties['animation'];
+	background:
+		| {
+				attachment: CSSProperties['backgroundAttachment'];
+				color: CSSProperties['backgroundColor'];
+				image: CSSProperties['backgroundImage'];
+				position: CSSProperties['backgroundPosition'] | number[]; // Can be written using array e.g. `[0 0]`
+				repeat: CSSProperties['backgroundRepeat'];
+				size: Array<CSSProperties['backgroundSize'] | CSSProperties['backgroundSize']>; // Can be written using array e.g. `['center' 'center']`
+			}
+		| CSSProperties['background'];
+	border:
+		| {
+				color: CSSProperties['borderColor'];
+				style: CSSProperties['borderStyle'];
+				width: CSSProperties['borderWidth'];
+			}
+		| CSSProperties['border'];
+	boxShadow:
+		| {
+				x: any;
```

Comment:
I think we can find more constrained types for these; e.g. aren't `x` and `y` constrained to length?

Reply:
Fixed in https://github.com/DefinitelyTyped/DefinitelyTyped/pull/24078/commits/8dda31f4f0a1e583cfc20159d0ec3fcacfc0b1c1